src/chart-addons/stacked-legend.js

```diff
@@ -98,7 +98,44 @@ function setColorScaleDomain_v1(domain) {
   }
 }
 
-export function getLegendStateFromChart(chart, useMap, selectedLayer) {
+async function getTopValues(layer, size) {
+  const OFFSET = 0
+  const dimension = layer?.getState()?.encoding?.color?.field
+  if (dimension) {
+    return layer
+      .crossfilter()
+      .dimension(dimension)
+      .group()
+      .topAsync(size, OFFSET)
+      .then(results => {
+        if (results) {
+          return results.map(result => result.key0)
+        } else {
+          return null
+        }
+      })
+      .catch(error => {
+        return null
```

Comment:
I wonder if we want to let this error bubble up instead of swallowing? That's assuming it will get caught in the immerse saga, and do the generic chart error.